docs/onion-message-channels.md

```diff
@@ -0,0 +1,220 @@
+# is hardcoded to 80):
+onion_serving_host = 127.0.0.1
+onion_serving_port = 8080
+
+# directory node configuration
+#
+# This is mandatory for directory nodes (who must also set their
+# own *.onion:port as the only directory in directory_nodes, below),
+# but NOT TO BE USED by non-directory nodes (which is you, unless
+# you know otherwise!), as it will greatly degrade your privacy.
+# (note the default is no value, don't replace it with "").
+hidden_service_dir =
+#
+# This is a comma separated list (comma can be omitted if only one item).
+# Each item has format host:port ; both are required, though port will
+# be 80 if created in this code.
+directory_nodes = rr6f6qtleiiwic45bby4zwmiwjrj3jsbmcvutwpqxjziaydjydkk5iad.onion:80
+
+# This setting is ONLY for developer regtest setups,
+# running multiple bots at once. Don't alter it otherwise
+regtest_count = 0,0
+```
+
+All of these can be left as default for most users, except the field `directory_nodes`.
+
+The list of **directory nodes** (the one shown here is one being run on signet, right now), which will
+be comma separated if multiple directory nodes are configured (we expect there will be 2 or 3 as a normal situation).
+The `onion_serving_port` is on which port on the local machine the onion service is served; you won't usually need to use it, but it mustn't conflict with some other usage (so if you have something running on port 8080, change it).
+The `type` field must always be `onion` in this case, and distinguishes it from IRC message channels and others.
+
+### Can/should I still run IRC message channels?
+
+In short, yes.
+
+### Do I need to configure Tor, and if so, how?
+
+To make outbound tor connections to other onions in the network, you will need to configure the
+SOCKS5 proxy settings (so, only directory nodes may *not* need this; everyone else does).
+This is identical to what we already do for IRC, except that in this case, we disallow clearnet connections.
+
+#### Running/testing as a maker
+
+A maker will additionally allow *inbound* connections to an onion service.
+This onion service will be ephemeral, that is, it will have a different onion address every time
+you restart. This should work automatically, using your existing tor daemon (here, we are using
+the same code as we use when running the `receive-payjoin` script, essentially).
+
+#### Running/testing as other bots (taker)
+
+A taker will not attempt to serve an onion; it will only use outbound connections, first to directory
+nodes and then, as according to need, to individual makers, also.
+
+As previously mentioned, both of these features - inbound and outbound, to onion, Tor connections - were already in use in Joinmarket. If you want to run/test as a maker bot, but never served an onion service before, it should work fine as long as you have the Tor service running in the background,
+and the default control port 9051 (if not, change that value in the `joinmarket.cfg`, see above.
```

Comment:
```suggestion
and the default control port 9051 (if not, change that value in the `joinmarket.cfg`, see above).
```
Missing parenthesis.

Reply:
Done in 4cff6d7bb1e3f81473c99dccd3460aae2f2a6c55